io/plugins/eu.esdihumboldt.hale.io.csv/src/eu/esdihumboldt/hale/io/csv/writer/AbstractTableInstanceWriter.java

```diff
@@ -58,8 +60,69 @@ public boolean isCancelable() {
 	protected Map<String, Object> getPropertyMap(Instance instance, List<String> headerRow,
 			boolean solveNestedProperties) {
 		// properties of current instance
-		Iterable<QName> allProperties = instance.getPropertyNames();
+		Iterable<QName> allProperties = DefinitionUtil.getAllProperties(instance.getDefinition())
```

Comment:
This looks like the method below is basically duplicated here. I think ideally would be to remove this one so it is always explicit if `useSchema` is enabled or not.

Reply:
OK